animation_nodes/nodes/gpencil/gp_material_output.py

```diff
@@ -0,0 +1,48 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... base_types import AnimationNode
+
+modeTypeItems = [
+    ("LINE", "Line", "Draw for strokes using a continuous line", "NONE", 0),
+    ("DOTS", "Dots", "Draw for strokes using separated dots", "NONE", 1),
+    ("BOX", "Boxes", "Draw for strokes using separated rectangle boxes", "NONE", 2)
+]
+
+class GPMaterialOutputNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GPMaterialOutputNode"
+    bl_label = "GP Material Output"
+
+    modeType: EnumProperty(name = "Mode Type", default = "LINE",
+        items = modeTypeItems, update = propertyChanged)
+
+    def create(self):
+        self.newInput("Material", "Material", "material", defaultDrawType = "PROPERTY_ONLY")
+        self.newInput("Boolean", "Stroke", "strokeBool", value = True)
+        self.newInput("Color", "Stroke Color", "strokeColor")
+        self.newInput("Boolean", "Fill", "fillBool", value = False)
+        self.newInput("Color", "Fill Color", "fillColor")
+        self.newInput("Integer", "Pass Index", "passIndex", value = 0, hide = True)
+
+        self.newOutput("Material", "Material", "material")
+
+    def draw(self, layout):
+        layout.prop(self, "modeType", text = "")
+
+    def execute(self, material, strokeBool, strokeColor, fillBool, fillColor, passIndex):
+        if material is None: return None
+        if not material.is_grease_pencil:
+            bpy.data.materials.create_gpencil_data(material)
+
+        gpMaterial = material.grease_pencil
+        gpMaterial.show_stroke = strokeBool
+        gpMaterial.mode = self.modeType
+        gpMaterial.show_fill = fillBool
+
+        if strokeBool and  gpMaterial.stroke_style == "SOLID":
```

Comment:
I don't see why we need those if statements. We should set the colors regardless.

Reply:
Omar, It is needed, when `stroke_style = 'TEXTURE'` then there no option for `color`.

Reply:
The color data is still there, and it should be set regardless. In fact, when the *Use As Stencil Mask* or the *Mix With Color* options are enabled in the `TEXTURE` style, the color will be utilized.

Reply:
You're right. Thanks.